modules/outbound-command-event-handler/src/domain/bulk_transaction_agg/index.ts

```diff
@@ -433,62 +434,65 @@ export class BulkTransactionAgg extends BaseAggregate<BulkTransactionEntity, Bul
-                const individualTransfer = await this.getIndividualTransferById(individualTransferId);
-                const party = individualTransfer.partyResponse?.party;
-                const condition = individualTransfer.quoteResponse?.condition;
-                if(!condition) {
-                    throw new Error(`condition for BulkBatch[${bulkBatchId}].IndividualTransfer[${individualTransferId}] is missing!`);
-                }
-                const ilpPacket = individualTransfer.quoteResponse?.ilpPacket;
-                if(!ilpPacket) {
-                    throw new Error(`ilpPacket for BulkBatch[${bulkBatchId}].IndividualTransfer[${individualTransferId}] is missing!`);
-                }
-
-                if(party) {
-                    // Generate Transfers request
-                    const individualBulkTransferRequest: SDKSchemeAdapter.V2_0_0.Outbound.Types.individualTransfer = {
-                        transferId: individualTransfer.id,
-                        to: {
-                            idType: party.partyIdInfo.partyIdType,
-                            idValue: party.partyIdInfo.partyIdentifier,
-                            idSubValue: party.partyIdInfo.partySubIdOrType,
-                            displayName: party.name,
-                            firstName: party.personalInfo?.complexName?.firstName,
-                            middleName: party.personalInfo?.complexName?.middleName,
-                            lastName: party.personalInfo?.complexName?.lastName,
-                            dateOfBirth: party.personalInfo?.dateOfBirth,
-                            merchantClassificationCode: party.merchantClassificationCode,
-                            fspId: party.partyIdInfo.fspId,
-                            extensionList: party.partyIdInfo.extensionList?.extension,
-                        },
-                        amountType: individualTransfer.request.amountType,
-                        currency: individualTransfer.request.currency,
-                        amount: individualTransfer.request.amount,
-                        condition,
-                        ilpPacket,
-                        transactionType: 'TRANSFER',
-                        extensions: individualTransfer.request.quoteExtensions,
-                    };
-                    this._logger.debug(`Generated Transfers Request ${JSON.stringify(individualBulkTransferRequest, null, 2)} for BulkBatch.bulkQuoteId=${bulkBatch.bulkQuoteId}`);
-                    // Add Transfers to batch
-                    bulkBatch.addIndividualTransfer(
-                        individualBulkTransferRequest,
-                        individualTransfer.id,
-                    );
+            if(bulkBatch.state == BulkBatchInternalState.AGREEMENT_COMPLETED) {
+                const individualQuoteResults = bulkBatch.bulkQuotesResponse?.individualQuoteResults; // TODO: Are we not missing the condition here?
```

Comment:
```suggestion
                const individualQuoteResults = bulkBatch.bulkQuotesResponse?.individualQuoteResults;
```

TODO comment is no longer valid since we deal with the condition on line 476.